presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -376,6 +377,20 @@ public void checkCanShowGrants(ConnectorTransactionHandle transaction, Identity
         return grantInfos;
     }
 
+    @Override
+    public void checkCanShowRoles(ConnectorTransactionHandle transactionHandle, Identity identity, String catalogName)
```

Comment:
Another option is that you have a checkCanAccessCatalog when you get the catalog id, which is a prereq to everything